carma_novatel_driver_wrapper/src/carma_novatel_driver_wrapper.cpp

```diff
@@ -0,0 +1,114 @@
+#include "carma_novatel_driver_wrapper/carma_novatel_driver_wrapper.hpp"
+#include <gps_msgs/msg/gps_fix.hpp>
+#include <sensor_msgs/msg/imu.hpp>
+#include <uncertainty_tools/uncertainty_tools.h>
+
+using std_msec = std::chrono::milliseconds;
+
+namespace carma_novatel_driver_wrapper
+{   
+    CarmaNovatelDriverWrapper::CarmaNovatelDriverWrapper(const rclcpp::NodeOptions &options)
+            : CarmaLifecycleNode(options)
+    {
+        
+    }
+    
+    void CarmaNovatelDriverWrapper::inspvax_callback(const novatel_oem7_msgs::msg::INSPVAX::UniquePtr msg)
+    {
+        last_gnss_msg_ = this->now();
+
+        // NOTE: At the moment CARMA's hardware interfaces do not use GPS status information
+        // Therefore that information is not currently populated in this message
+        // Convert position
+        gnss_fix_fused_msg_.header = msg->header;
+        gnss_fix_fused_msg_.latitude = msg->latitude;
+        gnss_fix_fused_msg_.longitude = msg->longitude;
+        gnss_fix_fused_msg_.altitude = msg->height;
+        //Set position covariance
+        gnss_fix_fused_msg_.position_covariance_type = gps_msgs::msg::GPSFix::COVARIANCE_TYPE_DIAGONAL_KNOWN;
+        gnss_fix_fused_msg_.position_covariance[0] = msg->longitude_stdev * msg->longitude_stdev;
+        gnss_fix_fused_msg_.position_covariance[4] = msg->latitude_stdev * msg->latitude_stdev;
+        gnss_fix_fused_msg_.position_covariance[8] = msg->height_stdev * msg->height_stdev;
+
+        //Convert orientation
+        // NOTE: It is unclear in the message spec what dip represents so it will be ignored
+        gnss_fix_fused_msg_.track = msg->azimuth;
+        gnss_fix_fused_msg_.pitch = msg->pitch;
+        gnss_fix_fused_msg_.roll = msg->roll;
+
+        // GPSFix messages request uncertainty reported with 95% confidence interval. That is 2 times the standard deviation
+        gnss_fix_fused_msg_.err_track = 2.0 * msg->azimuth_stdev;
+        gnss_fix_fused_msg_.err_pitch = 2.0 * msg->pitch_stdev;
+        gnss_fix_fused_msg_.err_roll = 2.0 * msg->roll_stdev;
+
+        // Convert Velocity
+        std::vector<double> components, variances;
+        components.push_back(msg->north_velocity);
+        variances.push_back(2.0 * msg->north_velocity_stdev);
+
+        components.push_back(msg->east_velocity);
+        variances.push_back(2.0 * msg->east_velocity_stdev);
+
+        std::tuple<double, double> speed_tuple = uncertainty_tools::computeVectorMagnitudeAndUncertainty(components, variances);
+        gnss_fix_fused_msg_.speed = std::get<0>(speed_tuple);
+        gnss_fix_fused_msg_.climb = msg->up_velocity;
+
+        gnss_fix_fused_msg_.err_climb = 2.0 * msg->up_velocity_stdev;
+        gnss_fix_fused_msg_.err_speed = std::get<1>(speed_tuple);
+
+        fix_fused_pub_->publish(gnss_fix_fused_msg_);
+    }
+
+    void CarmaNovatelDriverWrapper::imu_callback(const sensor_msgs::msg::Imu::UniquePtr msg)
+    {
+        last_imu_msg_ = this->now();
```

Comment:
You have these timeouts setup, but they don't seem to be used. What you should do, is if the timeout occurs then throw an exception to be caught. This will cause a system alert to be propagated to the subsystem lifecycle controller.